src/darker/__main__.py

```diff
@@ -572,9 +573,16 @@ def main(  # pylint: disable=too-many-locals,too-many-branches,too-many-statemen
     else:
         # In other modes, only reformat files which have been modified.
         if git_is_repository(root):
-            changed_files_to_reformat = git_get_modified_python_files(
-                files_to_process, revrange, root
-            )
+            # Get the modified files only.
+            repo_root = darker.black_compat.find_project_root([str(root)])
+            changed_files = {
+                (repo_root / file).relative_to(root)
+                for file in
+                git_get_modified_python_files(paths, revrange, repo_root)
```

Comment:
Sorry for bad formatting in my suggestion. Black actually wants to have it like this:
```suggestion
                for file in git_get_modified_python_files(paths, revrange, repo_root)
```